packages/ui/src/services/utils.ts

```diff
@@ -74,21 +74,22 @@ export const getValueByPath = <T extends Record<string, unknown>>(option: T, pro
 }
 
 /**
- * Finds value of nested property inside of an object.
+ * Finds value of nested property inside an object.
  *
- * @param option - Object to look properties inside
- * @param prop - string or function used to find nested property
+ * @param option - Object to look properties inside.
+ * @param prop - String or function used to find nested property.
  */
-export const getValueByKey = <T extends (string | number | Record<string, unknown>)> (
+export const getValueByKey = <T extends Record<string, unknown>> (
   option: T,
   prop: string | ((option: T) => any),
 ): any => {
-  // Can't access not object option
-  if (typeof option !== 'object' || !option) { return undefined }
+  if (!option || typeof option !== 'object' || Array.isArray(option)) { return undefined }
 
   if (!prop) { return option }
   if (typeof prop === 'string') { return getValueByPath(option, prop) }
   if (typeof prop === 'function') { return prop(option) }
+
+  // if `prop` has different to string or function type and can't be processed
   return option
```

Comment:
I think we should return `undefined` if we can't get the value by the key

Reply:
Discussed.